src/printer/widgets/position.js

```diff
@@ -0,0 +1,39 @@
+import { CM_PER_INCH } from '../../shared/constants';
+
+const WIDGET_OFFSET_MM = 4;
+
+const DEFAULT_POSITIONS = {
+  'north-arrow': 'top-right',
+  scalebar: 'bottom-left',
+  attributions: 'bottom-right',
+};
+
+/**
+ * Applies transforms to match the given widget position and dpi
+ * @param {CanvasRenderingContext2D} ctx
+ * @param {'north-arrow'|'scalebar'|'attributions'} widgetType
+ * @param {import('../../main/index').WidgetPosition} position
```

Comment:
This smells like circular dependency...

Reply:
Haha, well the typecheck passes so I wouldn't be worried! maybe there is a solution to avoid all this `import(...)` mess in the types though